diff --git a/Makefile.direct b/Makefile.direct
index 30c70c3..20fdb98 100644
--- a/Makefile.direct
+++ b/Makefile.direct
@@ -11,6 +11,8 @@
 # check-cpp - runs C++ test without adding C++ (and cords) interface to gc.a
 # cord/de - builds dumb editor based on cords.
 
+CROSS_PREFIX=
+
 ABI_FLAG=
 # ABI_FLAG should be the cc flag that specifies the ABI.  On most
 # platforms this will be the empty string.  Possible values:
@@ -21,19 +23,19 @@ AS_ABI_FLAG=$(ABI_FLAG)
 # ABI flag for assembler.  On HP/UX this is +A64 for 64 bit
 # executables.
 
-CC=cc $(ABI_FLAG)
+CC=$(CROSS_PREFIX)gcc $(ABI_FLAG)
 # Compiler executable name.  For EMX, replace to "gcc".
 
-CXX=g++ $(ABI_FLAG)
+CXX=$(CROSS_PREFIX)g++ $(ABI_FLAG)
 # Needed only for "make c++", which adds the c++ interface.
 
-AS=as $(AS_ABI_FLAG)
+AS=$(CROSS_PREFIX)as $(AS_ABI_FLAG)
 # The above doesn't work with gas, which doesn't run cpp.
 # Define AS as `gcc -c -x assembler-with-cpp' instead.
 # Under Irix 6, you have to specify the ABI (-o32, -n32, or -64)
 # if you use something other than the default ABI on your machine.
 
-LD=ld
+LD=$(CROSS_PREFIX)ld
 
 # Redefining srcdir allows object code for the nonPCR version of the collector
 # to be generated in different directories.
@@ -48,6 +50,7 @@ CFLAGS_EXTRA=
 CFLAGS= -O -I$(srcdir)/include -I$(AO_SRC_DIR)/src \
   -DALL_INTERIOR_POINTERS -DENABLE_DISCLAIM -DGC_ATOMIC_UNCOLLECTABLE \
   -DGC_GCJ_SUPPORT -DJAVA_FINALIZATION -DNO_EXECUTE_PERMISSION \
+  -DGC_THREADS \
   -DUSE_MMAP -DUSE_MUNMAP $(CFLAGS_FOR_PIC) $(CFLAGS_EXTRA)
 
 # To build the collector with threads support, add to the above:
@@ -63,7 +66,7 @@ CFLAGS= -O -I$(srcdir)/include -I$(AO_SRC_DIR)/src \
 # part of the build process, i.e. on the build machine.  These will usually
 # be the same as CC and CFLAGS, except in a cross-compilation environment.
 # Note that HOSTCFLAGS should include any -D flags that affect thread support.
-HOSTCC=$(CC)
+HOSTCC=gcc
 HOSTCFLAGS=$(CFLAGS)
 
 # For dynamic library builds, it may be necessary to add flags to generate
@@ -76,9 +79,9 @@ HOSTCFLAGS=$(CFLAGS)
 # influencing the collector configuration.
 
 CXXFLAGS= $(CFLAGS)
-AR= ar
+AR= $(CROSS_PREFIX)ar
 
-RANLIB= ranlib
+RANLIB= $(CROSS_PREFIX)ranlib
 # For EMX, replace "ranlib" with "ar s".
 
 
diff --git a/include/gc_config_macros.h b/include/gc_config_macros.h
index 14c65ee..07a7995 100644
--- a/include/gc_config_macros.h
+++ b/include/gc_config_macros.h
@@ -64,6 +64,9 @@
 #if defined(USE_LD_WRAP)
 # define GC_USE_LD_WRAP
 #endif
+#if defined(_WIN32)
+# define GC_WIN32_PTHREADS
+#endif
 
 #if defined(GC_WIN32_PTHREADS) && !defined(GC_WIN32_THREADS)
   /* Using pthreads-win32 library (or other Win32 implementation).  */
